### PR TITLE
Fix SIMDComparisonUtil build failure on aarch64

### LINKED ISSUE #10471: SIMDComparisonUtil.h:43:69: error: no matching function for call to ‘xsimd::batch_bool<signed char, xsimd::neon64>::batch_bool(xsimd::batch<signed char>)’
### Problem description

```
In file included from /prestissimo/velox/velox/functions/prestosql/Comparisons.cpp:20:
/prestissimo/velox/velox/functions/lib/SIMDComparisonUtil.h: In function ‘uint64_t facebook::velox::functions::detail::to64Bits(const int8_t*)’:
/prestissimo/velox/velox/functions/lib/SIMDComparisonUtil.h:43:69: error: no matching function for call to ‘xsimd::batch_bool<signed char, xsimd::neon64>::batch_bool(xsimd::batch<signed char>)’
   43 |         xsimd::batch_bool<int8_t>(d_type::load_unaligned(resultData)));
      |                                                                     ^
In file included from /prestissimo/_build/_deps/xsimd-src/include/xsimd/xsimd.hpp:61,
                 from /prestissimo/velox/velox/common/base/SimdUtil.h:24,
                 from /prestissimo/velox/velox/vector/FlatVector.h:23,
                 from /prestissimo/velox/velox/expression/StringWriter.h:23,
                 from /prestissimo/velox/velox/expression/CastHooks.h:19,
                 from /prestissimo/velox/velox/expression/CastExpr.h:19,
                 from /prestissimo/velox/velox/functions/prestosql/types/TimestampWithTimeZoneType.h:18,
                 from /prestissimo/velox/velox/functions/prestosql/Comparisons.h:20,
                 from /prestissimo/velox/velox/functions/prestosql/Comparisons.cpp:17:
/prestissimo/_build/_deps/xsimd-src/include/xsimd/types/xsimd_batch.hpp:306:9: note: candidate: ‘template<class Tp> xsimd::batch_bool<T, A>::batch_bool(const Tp*) [with Tp = Tp; T = signed char; A = xsimd::neon64]’ (deleted)
  306 |         batch_bool(Tp const*) = delete;
      |         ^~~~~~~~~~
/prestissimo/_build/_deps/xsimd-src/include/xsimd/types/xsimd_batch.hpp:306:9: note:   template argument deduction/substitution failed:
In file included from /prestissimo/velox/velox/functions/prestosql/Comparisons.cpp:20:
/prestissimo/velox/velox/functions/lib/SIMDComparisonUtil.h:43:69: note:   mismatched types ‘const Tp*’ and ‘xsimd::batch<signed char>’
   43 |         xsimd::batch_bool<int8_t>(d_type::load_unaligned(resultData)));
      |                                                                     ^
```

Introduced by this PR https://github.com/facebookincubator/velox/pull/10273

### System information

Velox System Info v0.0.2
Commit: 322d892f97b46aefcb1b40b5b34e016f316fa8e8
CMake Version: 3.28.3
System: Linux-6.7.7-200.fc39.aarch64
Arch: aarch64
C++ Compiler: /usr/bin/c++
C++ Compiler Version: 11.4.0
C Compiler: /usr/bin/cc
C Compiler Version: 11.4.0
CMake Prefix Path: /usr/local;/usr;/;/usr/local/lib/python3.10/dist-packages/cmake/data;/usr/local;/usr/X11R6;/usr/pkg;/opt
  

### CMake log

_No response_

### PR DESCRIPTION
**Pull request by zhli1142015**
Fixes #10471

### CODE REVIEW

**Comment by zhli1142015** [comment]
cc @majetideepak , thanks

**Review by majetideepak — APPROVED** [review]
I verified that this fixes the build issue. Thanks!

File: velox/functions/lib/SIMDComparisonUtil.h [hunk 1]
```diff
@@ -39,24 +39,28 @@ inline uint64_t to64Bits(const int8_t* resultData) {
       "Unsupported number of scalar elements");
   uint64_t res = 0UL;
   if constexpr (numScalarElements == 64) {
-    res = simd::toBitMask(
-        xsimd::batch_bool<int8_t>(d_type::load_unaligned(resultData)));
+    res = simd::toBitMask(xsimd::batch_bool<int8_t>(
+        simd::reinterpretBatch<uint8_t>(d_type::load_unaligned(resultData))));
```
[the remark this answers is not in the document]
Comment (zhli1142015) [reply velox/functions/lib/SIMDComparisonUtil.h:43]: `resultData` is already int8_t, `reinterpretBatch` function only supports int8_t -> uint8_t not int8_t  -> int8_t.
BTW, this fix is inspired by https://github.com/facebookincubator/velox/blob/0adc62e3aaf647596ab476f6300a4165e60f6b91/velox/dwio/common/ColumnVisitors.h#L887

[the remark this answers is not in the document]
Comment (majetideepak) [reply velox/functions/lib/SIMDComparisonUtil.h:43]: Thanks!

**Comment by majetideepak** [comment]
@Yuhta Can you take another look at this change?

**Review by Yuhta — APPROVED** [review]

**Pull request closed without merging**